### DISCUSSION TITLE
is Unsubscribe() required before closing connection? (PubSub)

### DISCUSSION THREAD

**Discussion by HeCorr** [redis/go-redis#2167] [Q&A]
Hello! I was wondering if I should call `Unsubscribe()` before closing the connection to Redis or if it doesn't really matter since the connection is going to be closed anyway.

Here's the relevant part of my code:

![image](https://user-images.githubusercontent.com/75134774/180885301-bcde1c69-f7a6-4a46-848e-0d198cd0f8e2.png)

I noticed that go-redis automatically creates a secondary connection for keepalive pings so I had to close both `rc` and `sub`, otherwise I would get errors that a closed connection was being used.

Please let me know if there's something flawed in my code. Thanks!

**Comment by bbrombacher** [comment]
I am also curious about the necessity of using sub.Unsubscribe().

Specifically, I have a websocket endpoint and currently I unsubscribe when the websocket connection closes.

But I'm wondering if the connection to Redis will automatically drop when the Go routine that handled the websocket request ends?
